### PR TITLE
Add change title functionality

### PR DESCRIPTION
**Pull request by JRojas999**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java [hunk 1]
```diff
@@ -23,52 +22,58 @@
...
+     * File Request Form stores parameters for an file request.
      */
     public FileRequestForm() {
     }
 
     /**
-     * Adds projId parameter
+     * Adds projId parameter.
+     *
      * @param projId
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java]: Add document about param projId

File: src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java [hunk 2]
```diff
@@ -23,52 +22,58 @@
...
      */
     public void addProjectId(final String projId) {
         addParameters(new Parameter(projectId, projId, false));
     }
     /**
-     * Adds title parameter
+     * Adds title parameter.
+     *
      * @param title
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java]: Add document about param title

File: src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java [hunk 3]
```diff
@@ -23,52 +22,58 @@
...
     public void addFileExtension(final String extension) {
         addParameters(new Parameter(fileExtension, extension, false));
     }
 
     /**
-     * Gets the body parameters
-     * @return bodyParameters
+     * Gets the body parameters.
+     *
+     * @return bodyParameters.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java]: Add document about  @return bodyParameters.

File: src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java [hunk 4]
```diff
@@ -77,8 +82,9 @@ public void addParameters(final Parameter parameter) {
     }
 
     /**
-     * Gets the url
-     * @return url
+     * Gets the url.
+     *
+     * @return url.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java]: Add document about @return url.

File: src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java [hunk 5]
```diff
@@ -23,52 +22,58 @@
     private final String fileExtension = "extension";
 
     /**
-     * File Request Form stores parameters for an file request
+     * File Request Form stores parameters for an file request.
      */
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java]: Delete Javadoc for constructor

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 6]
```diff
@@ -22,54 +22,54 @@
     private final String type = "type";
 
     /**
-     * Project Request Form stores parameters for an project request
+     * Project Request Form stores parameters for an project request.
      */
     public ProjectRequestForm() {
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java:26]: Delete Javadoc for constructor

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 7]
```diff
@@ -22,54 +22,54 @@
...
     }
 
     /**
-     * Adds id parameter
-     * @param id
+     * Adds id parameter.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java:30]: Add space line with *  between 31 and 32 line

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 8]
```diff
@@ -22,54 +22,54 @@
...
+     * @param id is project id.
      */
     public void addProjectId(final String id) {
         addParameters(new Parameter(projectId, id, false));
     }
     /**
-     * Adds name parameter
-     * @param name
+     * Adds name parameter.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java:39]: Add space line with * between 31 and 32 line

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 9]
```diff
@@ -22,54 +22,54 @@
...
+     * @param name is project name.
      */
     public void addProjectName(final String name) {
         addParameters(new Parameter(projectName, name, false));
     }
     /**
-     * Adds type parameter
-     * @param type
+     * Adds type parameter.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java:48]: Add space line between description and @param

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 10]
```diff
@@ -22,54 +22,54 @@
...
-    public void addType(final String type) {
-        addParameters(new Parameter(this.type, type, false));
+    public void addType(final String projectType) {
+        addParameters(new Parameter(type, projectType, false));
     }
 
     /**
-     * Gets the body parameters
-     * @return bodyParameters
+     * Gets the body parameters.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java:57]: Add space line between description and @return

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 11]
```diff
@@ -22,54 +22,54 @@
...
      */
     @Override
     public List<Parameter> getBodyParameters() {
         return bodyParameters;
     }
 
     /**
-     * Adds parameters to bodyParameters
-     * @param parameter
+     * Adds parameters to bodyParameters.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java:67]: Add space line between description and @param

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 12]
```diff
@@ -22,54 +22,54 @@
...
     @Override
     public void addParameters(final Parameter parameter) {
         bodyParameters.add(parameter);
     }
 
     /**
-     * Gets the url
-     * @return url
+     * Gets the url.
+     * @return url.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java]: Add space line between description and @return, and add description the last

File: src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java [hunk 13]
```diff
@@ -1,14 +1,13 @@
 /**
  * Copyright (c) 2021 Fundacion Jala.
- *
+ * <p>
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java]: delete <p>

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java [hunk 14]
```diff
@@ -72,23 +73,24 @@ public void insertUpdate(final DocumentEvent de) {
...
         textCodeArea.setRowHeaderView(lineCode);
         add(textCodeArea, BorderLayout.CENTER);
     }
 
     /**
-     * Gives us content of text area
-     * @return a String
+     * Gives us content of codeArea textarea.
+     *
+     * @return a string.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:96]: Add new blank line at the end of class

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java [hunk 15]
```diff
@@ -37,6 +36,7 @@
 
     /**
      * Initializes the graphic elements for Compiler buttons.
+     *
      * Runs code, cleans console, uploads code.
      */
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java:44]: Add new blank line at the end of class

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 16]
```diff
@@ -1,14 +1,13 @@
 /**
  * Copyright (c) 2021 Fundacion Jala.
- *
+ * <p>
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: Delete  <p>

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 17]
```diff
@@ -86,10 +85,10 @@ public CompilerInterface() {
         add(consoleOutput, panelConstraint);
         buttonsCompiler.getRunButton().addActionListener(new ActionListener() {
             @Override
-            public void actionPerformed(ActionEvent e) {
+            public void actionPerformed(final ActionEvent e) {
                 LOGGER.info("start");
                 String url = "";
-                if (!langButtons.getPython().isEnabled()){
+                if (!langButtons.getPython().isEnabled()) {
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:94]: Delete blank line in line 96

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 18]
```diff
@@ -120,14 +119,14 @@ public void actionPerformed(ActionEvent e) {
         });
         langButtons.getJava().addActionListener(new ActionListener() {
             @Override
-            public void actionPerformed(ActionEvent e) {
+            public void actionPerformed(final ActionEvent e) {
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:125]: Add blank line before  @Override

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 19]
```diff
@@ -120,14 +119,14 @@ public void actionPerformed(ActionEvent e) {
...
                 langButtons.getJava().setEnabled(false);
                 langButtons.getPython().setEnabled(true);
             }
         });
         langButtons.getPython().addActionListener(new ActionListener() {
             @Override
-            public void actionPerformed(ActionEvent e) {
+            public void actionPerformed(final ActionEvent e) {
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:132]: Add new blank line at the end of class

File: src/main/java/org/fundacion/jala/converter/view/compiler/Console.java [hunk 20]
```diff
@@ -39,6 +38,7 @@ protected Console() {
 
     /**
      * Gets access to JTextArea
+     *
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/Console.java:44]: Add new blank line at the end of class

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java [hunk 21]
```diff
@@ -54,6 +53,7 @@ public String getText() {
 
             /**
              * Adds the number of rows in Code Area Text.
+             *
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:59]: Delete Javadoc for constructor

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java [hunk 22]
```diff
@@ -37,6 +36,7 @@
 
     /**
      * Initializes the graphic elements for Compiler buttons.
+     *
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java:42]: Delete Javadoc for constructor

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 23]
```diff
@@ -50,11 +48,12 @@ public CompilerInterface() {
         langButtons.getJava().setEnabled(false);
         ProjectTab projectTab = new ProjectTab();
         projectTab.setFont(new Font("Barlow", 0, 11));
+        codeArea.setName("Main");
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:54]: Delete Javadoc for constructor

File: src/main/java/org/fundacion/jala/converter/view/compiler/Console.java (hunk 20 above)
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/Console.java:44]: Delete Javadoc for constructor

File: src/main/java/org/fundacion/jala/converter/view/compiler/LanguageButtons.java [hunk 24]
```diff
@@ -54,6 +53,7 @@ protected LanguageButtons() {
 
     /**
      * Button that defines or not java is selected
+     *
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/LanguageButtons.java:59]: Delete Javadoc for constructor

File: src/main/java/org/fundacion/jala/converter/view/compiler/LanguageButtons.java [hunk 25]
```diff
@@ -62,6 +62,7 @@ public JButton getJava() {
 
     /**
      * Button that defines or not python is selected
+     *
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/LanguageButtons.java:68]: Add new blank line at the end of class

File: src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java [hunk 26]
```diff
@@ -20,20 +19,23 @@
...
         Line2D line2 = new Line2D.Double(new Point2D.Double(0, getHeight() / 2),
                 new Point2D.Double(getWidth(), getHeight() / 2));
-        g2.setStroke(new BasicStroke(3));
-        g2.draw(line1);
-        g2.draw(line2);
+        graphics2D.setStroke(new BasicStroke(LINE_WIDTH));
+        graphics2D.draw(line1);
+        graphics2D.draw(line2);
     }
 }
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java:44]: Add new blank line at the end of class

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 27]
```diff
@@ -8,65 +8,65 @@
...
         setSelectedIndex(getTabCount() - 2);
     }
 
     /**
-     * Deletes a selected tab
-     * @param title of tab we want to delete
-     * @return a boolean that means if works or not
+     * Deletes a selected tab.
+     *
+     * @param title of tab we want to delete.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:69]: Add black line

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 28]
```diff
@@ -79,10 +79,52 @@ public boolean removeTap(String title) {
...
+            String result = (String) JOptionPane.showInputDialog(null,
+                    "Change TabName", "Change name",
+                    JOptionPane.PLAIN_MESSAGE, null, null, "Red");
+            if (result != null && result.length() > 0) {
+                int indexTab = getSelectedIndex();
+                setTitleAt(indexTab, result);
+                setTabComponentAt(indexTab, makeTabTitle(result));
+            }
+        }
+    }
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:132]: Add new blank line at the end of class

File: src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java (hunk 5 above)
Comment (Raymundogs) [reply src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java]: the javadoc should go on the class.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 29]
```diff
@@ -8,65 +8,65 @@
...
+     *
+     * @param event means click on "button".
      */
     @Override
-    public void actionPerformed(ActionEvent e) {
-        contador ++;
-        CloseButton tabButton = new CloseButton();
-        String title= "EndGame " + contador;
+    public void actionPerformed(final ActionEvent event) {
+        COUNTER++;
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:54]: COUNTER should be a variable, not a constant.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 30]
```diff
@@ -8,65 +8,65 @@
...
-import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.ActionEvent;
 
 public class ProjectTab extends JTabbedPane implements ActionListener {
-    public static int contador;
-    PlusButton button;
+    public static int COUNTER;
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:29]: COUNTER is not a constant, it should be written with lower camel case.

**Review by Paola-Xaq — COMMENTED** [review]
JavaDoc is missing for every class.

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 31]
```diff
@@ -79,10 +79,52 @@ public boolean removeTap(String title) {
...
+
+    /**
+     * Changes only name of title.
+     *
+     * @param numberOfClick represents how many clicks it receives.
+     */
+    public void changeTitle(final int numberOfClick) {
+        if (numberOfClick == 2 && !"Main".equals(getTitleAt(getSelectedIndex()))) {
+            String result = (String) JOptionPane.showInputDialog(null,
+                    "Change TabName", "Change name",
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:124]: It's only one parameter pee line.

File: src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java [hunk 32]
```diff
@@ -22,63 +21,69 @@
...
-    /**
-     * File Request Form stores parameters for an file request
-     */
     public FileRequestForm() {
     }
 
     /**
-     * Adds projId parameter
-     * @param projId
+     * Adds projId parameter.
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/FileRequestForm.java:31]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java [hunk 33]
```diff
@@ -8,7 +8,6 @@
  *
  * @author Joel Rodrigo Rojas Roman
  */
-
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/Models/ProjectRequestForm.java:11]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java [hunk 34]
```diff
@@ -20,21 +19,24 @@
 import java.awt.geom.Point2D;
 
 class CloseButton extends JButton {
+    private final int LINE_WIDTH = 3;
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java:25]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java [hunk 35]
```diff
@@ -8,7 +8,6 @@
  *
  * @author Paola Aguilar Quiñones
  */
-
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:11]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 36]
```diff
@@ -21,7 +20,6 @@
 import org.apache.http.util.EntityUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
-
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:24]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/Console.java [hunk 37]
```diff
@@ -8,7 +8,6 @@
  *
  * @author Paola Aguilar Quiñones
  */
-
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/Console.java:11]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/LanguageButtons.java [hunk 38]
```diff
@@ -8,7 +8,6 @@
  *
  * @author Paola Aguilar Quiñones
  */
-
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/LanguageButtons.java:11]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java [hunk 39]
```diff
@@ -8,7 +8,6 @@
  *
  * @author Joel Rodrigo Rojas Roman
  */
-
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java:11]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 40]
```diff
@@ -8,65 +8,65 @@
...
 import javax.swing.JLabel;
+import javax.swing.JOptionPane;
 import java.awt.Dimension;
 import java.awt.FlowLayout;
 import java.awt.Font;
-import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.ActionEvent;
```
Comment (GustavoZHA) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:23]: add javadoc document in class

File: src/main/java/org/fundacion/jala/converter/view/compiler/Console.java (hunk 20 above)
Comment (GustavoZHA) [reply src/main/java/org/fundacion/jala/converter/view/compiler/Console.java:44]: Line 23 and 22 , width  and height  have to be uppercase

**Pull request closed without merging**